distributed/tests/test_steal.py

```diff
@@ -825,44 +825,6 @@ def block_reduce(x, y, event):
     assert not b.data
```

Comment:
`test_do_not_steal_communication_heavy_tasks` doesn't look right, but maybe I'm missing something?

Firstly, there's a race condition. The test waits for x and y to enter running state on the worker, polling every 100ms. 
The first iteration of `while not a.state.tasks` will always fail (because we never yielded the event loop snce submit).
The second iteration, 100ms later, *most times* will find that a and b are in memory and two block_reduce are executing. However there's nothing guaranteeing it, so when you call `steal.balance()` on the next line you might have x or y still running, or in memory but without the scheduler knowing yet. Unlikely, but possible.

The test implicitly relies on tasks of unknown duration to be stolen (#5572). It should be changed not to rely on this specific use case.

Finally, why bother calling `steal.stop()`? If the block_reduce tasks can't be stolen, it should be inconsequential.

Reply:
Overhauled test in #7250. This PR can be merged as is.